src/wallet/wallet.cpp

```diff
@@ -3829,17 +3829,13 @@ CWallet* CWallet::CreateWalletFromFile(const std::string walletFile)
 
     if (fFirstRun)
     {
-        // Create new keyUser and set as default key
-        if (gArgs.GetBoolArg("-usehd", DEFAULT_USE_HD_WALLET) && !walletInstance->IsHDEnabled()) {
```

Comment:
After discussion I believe the removal of the !IsHDEnabled check here is fine but I could totally see it breaking something very subtly, so would be good to get more eyeballs on this line.

Reply:
IMO the `IsHDEnabled()` check was there because you could salvage the wallet which may then would have triggered a `fFirstRun` and recreated a HD master key. But this is no longer true since `fFirstRun` is always false if there is a key in the wallet (and HD master key is also a standard key).